### PR TITLE
Use grunt-exec module

### PR DESCRIPTION
**Pull request by markhalliwell**
Helps improve handling of external command executions (especially safari builds).

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -259,22 +259,46 @@ module.exports = function(grunt) {
...
   grunt.loadNpmTasks('grunt-contrib-less');
+  grunt.loadNpmTasks('grunt-contrib-qunit');
   grunt.loadNpmTasks('grunt-contrib-watch');
   grunt.loadNpmTasks('grunt-css2js');
+  grunt.loadNpmTasks('grunt-exec');
   grunt.loadNpmTasks('grunt-mozilla-addon-sdk');
-  grunt.loadNpmTasks('grunt-sed');
-  grunt.loadNpmTasks('grunt-contrib-qunit');
   grunt.loadNpmTasks('grunt-release');
+  grunt.loadNpmTasks('grunt-sed');
```
Comment (markhalliwell) [inline Gruntfile.js:301]: Just alphabetizing these.

File: Gruntfile.js [hunk 2]
```diff
@@ -259,22 +259,46 @@ module.exports = function(grunt) {
...
+        },
+        callback: function (error) {
+          if (error) {
+            grunt.log.warn('Unable to create ' + String(grunt.template.process('release/safari/<%= pkg.name %>.safariextension')).red);
+          }
+          else {
+            grunt.log.writeln('Created ' + String(grunt.template.process('release/safari/<%= pkg.name %>.safariextension')).cyan);
+          }
+        }
+      }
```
Comment (sun) [inline Gruntfile.js:284]: According to the docs https://github.com/jharding/grunt-exec#properties, couldn't this be simplified now?

File: Gruntfile.js [hunk 3]
```diff
@@ -324,23 +348,8 @@ module.exports = function(grunt) {
...
-    },
-    function (error, result, code) {
-      if (0 !== code) {
-        grunt.log.errorlns(result.stdout);
-        grunt.log.errorlns(result.stderr);
-      }
-    });
-  });
+  grunt.registerTask('build:safari', 'Builds the Safari extension.',
+    ['exec:build_safari_ext']);
```
Comment (sun) [inline Gruntfile.js:352]: Wondering whether this shouldn't actually be part of `release:safari`?  Do you need to sign a .safariextension if you just want to load a local dev build into your local Safari browser?

I guess that clarifies your recent question/suggestion of whether we shouldn't rename (or split) the current `build` tasks... — FF just compiles + builds/archives, Chrome does the same, only `build:safari` additionally tries to sign the extension to prepare it for a published release.

File: Gruntfile.js (hunk 1 above)
Comment (sun) [inline Gruntfile.js:301]: "compress" comes before "concat" in the alphabet though :8ball:

File: Gruntfile.js (hunk 3 above)
Comment (markhalliwell) [inline Gruntfile.js:352]: Yes, this is a requirement for .safariextz files to work. The build task create binaries, normal grunt builds dev folders (that can be symlinked). FF doesn't require signing for it to actually be install able. Safari does.

Ultimately FF should also be signed so we can host this on dreditor.org and auto update. But that's a separate issue. 

Typos courtesy of Siri

File: Gruntfile.js (hunk 2 above)
Comment (markhalliwell) [inline Gruntfile.js:284]: Simplified to what exactly??

Typos courtesy of Siri

File: Gruntfile.js (hunk 1 above)
Comment (markhalliwell) [inline Gruntfile.js:301]: Yep. It was late. 

Typos courtesy of Siri

**Pull request merged**